src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php

```diff
@@ -264,9 +264,9 @@ public function endTest($test, $time)
             unlink($this->runsInSeparateProcess);
             foreach ($deprecations ? unserialize($deprecations) : array() as $deprecation) {
                 if ($deprecation[0]) {
-                    trigger_error($deprecation[1], E_USER_DEPRECATED);
+                    trigger_error(serialize(array('deprecation' => $deprecation[1], 'class' => $deprecation[2], 'method' => $deprecation[3])), E_USER_DEPRECATED);
```

Comment:
I think the class and method here need to be determined by $test and not what's in $deprecation

Reply:
This code happens to work for \Symfony\Bridge\PhpUnit\Tests\ProcessIsolationTest() because the deprecation takes place in the test method. However, that won't be the case for a deprecation occurring in non test code.

Reply:
Well, we don't want to show the code triggering the deprecation. But we want to display a summary of tests that lead to these deprecations. Because you either need to fix the code executed by these tests or you will have to add these tests to the `legacy` group.

Reply:
Yes I totally agree but that's not what is in $deprecation[2] and $deprecation[3]. These contain the code that has triggered the deprecation not the test. That's in $test.

Reply:
I don't think so. We filter the stack trace before serealizing the same way as we do when not running tests in isolation. Do you have an example that wouldn't work?

Reply:
Ah I see I missed that :( sorry. I've tested with a deprecation in another file and you're correct that the stack trace filtering works. I kinda think that using $test is simpler but you're right it works.